### PR TITLE
Use pre-commit bot

### PR DESCRIPTION
**Pull request by jarrodmillman**
## Description

See #7172 

The bot is faster than the GH action. It provides the option of autofixing PR linting errors with a comment on the PR.  And it also will create PRs to update the pre-commit repos.

## Checklist

- A descriptive but concise pull request title
- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [Unit tests](https://scikit-image.org/docs/dev/development/contribute.html#testing)
- A gallery example in `./doc/examples` for new features
- [Contribution guide](https://scikit-image.org/docs/dev/development/contribute.html) is followed

## Release note

Summarize the introduced changes in the code block below in one or a few sentences. The
summary will be included in the next release notes automatically:

```release-note
...
```

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -41,3 +41,12 @@ repos:
         language: system
         entry: python tools/generate_requirements.py
         files: "pyproject.toml|requirements/.*\\.txt|tools/generate_requirements.py"
+
+ci:
+  # This ensures that pr's aren't autofixed by the bot, rather you call
```
Comment (mkcor) [inline .pre-commit-config.yaml:46]:
```suggestion
  # This ensures that PR's aren't autofixed by the bot, rather you call
```
(just more readable)

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -41,3 +41,12 @@ repos:
...
+  # the bot to make the fix
+  autofix_prs: false
+  autofix_commit_msg: |
+    '[pre-commit.ci 🤖] Apply code format tools to PR'
+  # Update hook versions every month (so we don't get hit with weekly update pr's)
```
Comment (mkcor) [inline .pre-commit-config.yaml:51]:
```suggestion
  # Update hook versions every month (so we don't get hit with weekly update PR's)
```

**Review by mkcor — APPROVED** [review]
Yay!

**Review by lagru — APPROVED** [review]
So we just need to comment "pre-commit.ci autofix" to our comment and then it goes and does the fix? There is no automated action besides a check that appears in the list of other checks?

In that case :+1:

**Comment by jarrodmillman** [comment]
And the bot will provide a report when there is an issue and tell you that you can fix it by adding a "pre-commit.ci autofix" comment. So no need to document it ourselves.

**Comment by lagru** [comment]
> And the bot will provide a report when there is an issue and tell you that you can fix it by adding a "pre-commit.ci autofix"

So is it creating a comment like pep8speaks does?

**Comment by jarrodmillman** [comment]
It will produce a `red x` like
2023-09-28T09:43:41,404078317-07:00

If you click on the `Details` for the `pre-commit.ci ...`, you are taken to:
https://results.pre-commit.ci/run/github/327679346/1693837106.vMDsW5paTUS464GkvaxVHA

That pages has been slightly updated, since I did what it said. It previously said to do this
https://github.com/scientific-python/specs/pull/264#issuecomment-1705350089

That produced this commit:
https://github.com/scientific-python/specs/pull/264/commits/6a05b2386bb7fdb0b61cca1066a9dfc62c0320ab

**Comment by lagru** [comment]
That UX seems really cool. :rocket:

**Pull request merged**